src/library-authoring/author-library/BlockPreview.jsx

```diff
@@ -0,0 +1,279 @@
+  showDeleteModal: PropTypes.bool.isRequired,
+  setShowDeleteModal: PropTypes.func.isRequired,
+  deleteLibraryBlock: PropTypes.func.isRequired,
+  previewKey: PropTypes.string.isRequired,
+  fetchBlockLtiUrl: PropTypes.func.isRequired,
+  isLtiUrlGenerating: PropTypes.bool,
+};
+
+BlockPreviewBase.defaultProps = {
+  isLtiUrlGenerating: false,
+};
+
+const BlockPreview = injectIntl(BlockPreviewBase);
+
+const inStandby = ({ blockStates, id, attr }) => blockStates[id][attr].status === LOADING_STATUS.STANDBY;
+const needsView = ({ blockStates, id }) => inStandby({ blockStates, id, attr: 'view' });
+const needsMeta = ({ blockStates, id }) => inStandby({ blockStates, id, attr: 'metadata' });
+
+/**
+ * BlockPreviewContainerBase
+ * Container component for the BlockPreview cards.
+ * Handles the fetching of the block view and metadata.
+ */
+const BlockPreviewContainerBase = ({
+  intl, block, blockView, blockStates, showPreviews, library, ltiUrlClipboard, ...props
+}) => {
+  // There are enough events that trigger the effects here that we need to keep track of what we're doing to avoid
+  // doing it more than once, or running them when the state can no longer support these actions.
+  //
+  // This problem feels like there should be some way to generalize it and wrap it to avoid this issue.
+  useEffect(() => {
+    props.initializeBlock({
+      blockId: block.id,
+    });
+  }, []);
+  useEffect(() => {
+    if (!blockStates[block.id] || !showPreviews) {
+      return;
+    }
+    if (needsMeta({ blockStates, id: block.id })) {
+      props.fetchLibraryBlockMetadata({ blockId: block.id });
+    }
+    if (needsView({ blockStates, id: block.id })) {
+      props.fetchLibraryBlockView({
+        blockId: block.id,
+        viewSystem: XBLOCK_VIEW_SYSTEM.Studio,
+        viewName: 'student_view',
+      });
+    }
+  }, [blockStates[block.id], showPreviews]);
+
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  // Need to force the iframe to be different if navigating away. Otherwise landing on the edit page
+  // will show the student view, and navigating back will show the edit view in the block list. React is smart enough
+  // to guess these iframes are the same between routes and will try to preserve rather than rerender, but that works
+  // against us here. Setting an explicit key prevents it from matching the two.
+  const previewKey = useMemo(() => `${uuid4()}`, [block.id]);
+
+  if (blockStates[block.id] === undefined) {
+    return <LoadingPage loadingMessage={intl.formatMessage(messages['library.detail.loading.message'])} />;
+  }
+  const { metadata } = blockStates[block.id];
+  const canEdit = metadata !== null && !BLOCK_TYPE_EDIT_DENYLIST.includes(metadata.block_type);
+
+  let editView;
+  if (canEdit) {
+    editView = ROUTES.Block.EDIT_SLUG(library.id, block.id);
+  } else {
+    editView = ROUTES.Detail.HOME_SLUG(library.id, block.id);
+  }
+
+  let isLtiUrlGenerating;
+  if (library.allow_lti) {
```

Comment:
```suggestion
  const isBlockOnClipboard = ltiUrlClipboard.value.blockId === block.id;
  if (library.allow_lt && iisBlockOnClipboard && ltiUrlClipboard.status === LOADING_STATUS.LOADED) {
      const clipboard = document.createElement('textarea');
      clipboard.value = getConfig().STUDIO_BASE_URL + ltiUrlClipboard.value.lti_url;
      document.body.appendChild(clipboard);
      clipboard.select();
      document.execCommand('copy');
      document.body.removeChild(clipboard);
    }

```
Probably it will simplify the statement

Reply:
good suggestion, but i think it breaks the logic, left as it was